src/index.ts

```diff
@@ -93,22 +149,28 @@ class DebuggerHandler<
   async update<W extends ConsolePanel | NotebookPanel | FileEditor>(
     debug: IDebugger,
     widget: W
-  ) {
+  ): Promise<void> {
     const debuggingEnabled = await debug.requestDebuggingEnabled();
-    if (!debug.model || this.handlers[widget.id] || !debuggingEnabled) {
+    if (
+      !debug.model ||
+      this.handlers[widget.id] ||
+      !debuggingEnabled ||
+      !debug.isStarted()
```

Comment:
I added  !debug.isStarted() due to this error 
![issue2](https://user-images.githubusercontent.com/34033913/70818458-0d33eb00-1dd4-11ea-8820-3bd60ee56af8.gif)

before update or create handler always is stared before setDebugSession where we restore and explicitly set debug session starts.

Reply:
Was this reverted in the last rebase?

Reply:
OK the code was restructured slightly it seems.